src/Val.cc

```diff
@@ -3292,18 +3632,21 @@ void VectorVal::ValDescribe(ODesc* d) const
 
 	size_t vector_size = vector_val->size();
 
-	if ( vector_size != 0)
+	if ( vector_size != 0 )
 		{
-		for ( unsigned int i = 0; i < (vector_size - 1); ++i )
+		auto last_ind = vector_size - 1;
+		for ( unsigned int i = 0; i < last_ind; ++i )
 			{
-			if ( vector_val->at(i) )
-				vector_val->at(i)->Describe(d);
+			auto v = At(i);
```

Comment:
Good to know.  (Note though that here there are two separate things going on, `at` vs `At`, so `v` wouldn't be appropriate for the first.)  So is `v`'s scope the remainder of the block, e.g., it could be used in an `else` branch too?